src/hotspot/os/posix/os_posix.cpp

```diff
@@ -1537,7 +1537,13 @@ void PlatformEvent::park() {       // AKA "down()"
   guarantee(_event >= 0, "invariant");
 }
 
-int PlatformEvent::park(jlong millis) {
+int PlatformEvent::park_millis(jlong millis) {
+  return park_nanos(millis_to_nanos_bounded(millis));
+}
+
+int PlatformEvent::park_nanos(jlong nanos) {
+  assert(0 <= nanos, "nanos are in range");
```

Comment:
`nanos` should never be zero else you call the untimed park.

Reply:
OK, I see how is that guaranteed in the Windows case. In POSIX case, calling `park()` is untimed wait, but `park(0)` is converted to absolute time that is already passed, and so `pthread_cond_timedwait` would return immediately, right? So `park(0)` is not equivalent to just `park()`? Still, the strongest behavior from Windows case takes precedence here. Changed the assert.